cypress/integration/purchase/purchase_order_complete_change_complete.js

```diff
@@ -0,0 +1,170 @@
+        .setName(packingMaterialName)
+        .setProduct(productForPackingMaterial)
+        .apply();
+    });
+    cy.fixture('product/packing_instructions.json').then(packingInstructionsJson => {
+      Object.assign(new PackingInstructions(), packingInstructionsJson)
+        .setName(packingInstructionsName)
+        .apply();
+    });
+    cy.fixture('product/packing_instructions_version.json').then(pivJson => {
+      Object.assign(new PackingInstructionsVersion(), pivJson)
+        .setName(packingInstructionsName)
+        .setPackingInstructions(packingInstructionsName)
+        .setPackingMaterial(packingMaterialName)
+        .apply();
+    });
+    cy.fixture('product/simple_productCategory.json').then(productCategoryJson => {
+      Object.assign(new ProductCategory(), productCategoryJson)
+        .setName(productCategoryName)
+        .setValue(productCategoryValue)
+        .apply();
+    });
+
+    Builder.createBasicProductEntitiesWithCUTUAllocation(
+      productCategoryName,
+      productCategoryValue,
+      priceListName,
+      productName1,
+      productValue1,
+      productType,
+      packingInstructionsName
+    );
+    cy.fixture('sales/simple_vendor.json').then(vendorJson => {
+      new BPartner({ name: vendorName })
+        .setVendor(true)
+        .setVendorPricingSystem(priceSystemName)
+        .setVendorDiscountSchema(discountSchemaName)
+        .setPaymentTerm('30 days net')
+        .addLocation(new BPartnerLocation('Address1').setCity('Cologne').setCountry('Deutschland'))
+        .apply();
+    });
+    cy.readAllNotifications();
+  });
+  it('Create a purchase order and complete it', function() {
+    cy.visitWindow('181', 'NEW');
+    cy.get('#lookup_C_BPartner_ID input')
+      .type(vendorName)
+      .type('\n');
+    cy.contains('.input-dropdown-list-option', vendorName).click();
+
+    cy.selectInListField('M_PricingSystem_ID', priceSystemName, false, null, true);
+    const addNewText = Cypress.messages.window.batchEntry.caption;
+    cy.get('.tabs-wrapper .form-flex-align .btn')
+      .contains(addNewText)
+      .should('exist')
+      .click();
+    cy.get('.quick-input-container .form-group').should('exist');
+    cy.writeIntoLookupListField('M_Product_ID', productName1, productName1, false, false, null, true);
+
+    cy.get('.form-field-Qty')
+      .click()
+      .find('.input-body-container.focused')
+      .should('exist')
+      .find('i')
+      .eq(0)
+      .click();
+    cy.server();
+    cy.route('POST', `/rest/api/window/${purchaseOrders.windowId}/*/${purchaseOrders.orderLineTabId}/quickInput`).as(
+      'resetQuickInputFields'
+    );
+    cy.get('.form-field-Qty')
+      .find('input')
+      .should('have.value', '0.1')
+      .clear()
+      .type('5{enter}');
+    cy.wait(8000);
+    /**Complete purchase order */
+    cy.get('.form-field-DocAction ul')
+      .click({ force: true })
+      .get('li')
+      .eq('1')
+      .click({ force: true });
+    cy.wait(8000);
```

Comment:
again, i *think* we can wait for the DocStatus to be "Completed", instead of waiting a static 8 seconds

Reply:
For some of the tests I couldn't find an alternative for the waits, need help/more expertise for this. In order to tackle all the waits we have in our tests, maybe we should create a follow-up task. There are multiple situations where we need to wait for background processes.